com.unity.ml-agents.extensions/Runtime/Input/InputActuatorComponent.cs

```diff
@@ -0,0 +1,376 @@
+        internal static IActuator[] GenerateActionActuatorsFromAsset(
+            InputActionMap defaultMap,
+            string layoutName,
+            uint localId,
+            BehaviorParameters behaviorParameters)
+        {
+            // TODO does this need to change based on the action map we use?
+            var builder = new InputControlLayout.Builder()
+                .WithName(layoutName)
+                .WithFormat(mlAgentsLayoutFormat);
+
+            var actuators = new List<IActuator>();
+
+            foreach (var action in defaultMap)
+            {
+                var actionLayout = InputSystem.LoadLayout(action.expectedControlType);
+
+                builder.AddControl(action.name)
+                    .WithLayout(action.expectedControlType)
+                    .WithFormat(actionLayout.stateFormat)
+                    .WithUsages($"MLAgentPlayer-{localId}");
+
+                var devicePath = InputControlPath.Separator + layoutName;
+
+                // Reasonably, the input system starts adding numbers after the first none numbered name
+                // is added.  So for device ID of 0, we use the empty string in the path.
+                var deviceId = localId == 0 ? string.Empty : "" + localId;
+                var path = $"{devicePath}{deviceId}{InputControlPath.Separator}{action.name}";
+                action.AddBinding(path,
+                    action.interactions,
+                    action.processors,
+                    mlAgentsControlSchemeName + localId);
+
+                var adaptor = (IRLActionInputAdaptor)Activator.CreateInstance(
+                    controlTypeToAdaptorType[actionLayout.type]);
+                actuators.Add(new InputActionActuator(
+                    behaviorParameters,
+                    action,
+                    adaptor));
+            }
+            var layout = builder.Build();
+            if (InputSystem.LoadLayout(layout.name) == null)
+            {
+                InputSystem.RegisterLayout(layout.ToJson(), layout.name);
+            }
+            return actuators.ToArray();
+        }
+
+        internal void FindNeededComponents()
+        {
+            if (m_InputAsset == null)
+            {
+                var assetProvider = GetComponent<IInputActionAssetProvider>();
+                Assert.IsNotNull(assetProvider);
+                (m_InputAsset, m_InputAssetCollection) = assetProvider.GetInputActionAsset();
+                Assert.IsNotNull(m_InputAsset, "An InputActionAsset could not be found on IInputActionAssetProvider or PlayerInput.");
+                Assert.IsNotNull(m_InputAssetCollection, "An IInputActionCollection2 could not be found on IInputActionAssetProvider or PlayerInput.");
+            }
+            if (m_PlayerInput == null)
+            {
+                m_PlayerInput = GetComponent<PlayerInput>();
+                Assert.IsNotNull(m_PlayerInput, "PlayerInput component could not be found on this GameObject.");
```

Comment:
Can we have a RequireComponent attribute on this component?

Reply:
sure